lib/v2/penguin-random-house/utils.js

```diff
@@ -0,0 +1,98 @@
+const cheerio = require('cheerio');
+const got = require('@/utils/got');
+const { art } = require('@/utils/render');
+const path = require('path');
+const dateParser = require('@/utils/dateParser');
+
+const parseBookInList = (element) => {
+    const $ = cheerio.load(element);
+    const title = $('h2').first().text();
+    const author = $('h2.author').first().text();
+    const description = $('.desc').first().text();
+    let imageSrc = $('img.cover__backcover').attr('src');
+    let imageAlt = $('img.cover__backcover').attr('alt');
+    if (!imageSrc) {
+        imageSrc = $('img.img-responsive').attr('data-src');
+        imageAlt = $('img.img-responsive').attr('alt');
+    }
+
+    return art(path.join(__dirname, 'templates/book.art'), {
+        title,
+        author,
+        description,
+        imageSrc,
+        imageAlt,
+    });
+};
+
+const parseBooks = (element) => {
+    const $ = cheerio.load(element);
+    const description = $('h2.read-down-text').first().html();
+    let mainBlock = '';
+
+    $('.awesome-list>li').map((i, element) => {
+        const appending = parseBookInList(element);
+        mainBlock += appending;
+        return appending;
+    });
+
+    return {
+        description,
+        content: mainBlock,
+    };
+};
+
+const parseArticle = (element) => {
+    const $ = cheerio.load(element);
+    const description = $('h2.hdr-smalltxt').first().html();
+    const imageSrc = $('div.img-block>img').first().attr('src');
+    const imageAlt = $('div.img-block>img').first().attr('alt');
+    let mainBlock = '';
+
+    const descriptionBlock = art(path.join(__dirname, 'templates/book.art'), {
+        description,
+        imageSrc,
+        imageAlt,
+    });
+
+    $('div.main-content>p,div.main-content>ul').map((i, element) => {
+        const appending = cheerio.load(element);
+        mainBlock += appending.html();
+        return appending;
+    });
+
+    return {
+        description: descriptionBlock,
+        content: mainBlock,
+    };
+};
+
+const parseList = async (items, ctx, contentParser) =>
+    await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.url, async () => {
+                const itemRes = await got({
+                    method: 'get',
+                    url: item.url,
+                });
```

Comment:
```suggestion
                const itemRes = await got(item.url);
```
https://github.com/DIYgod/RSSHub/pull/10287#discussion_r928079638